src/CallHandler.tsx

```diff
@@ -0,0 +1,482 @@
+                this.setCallState(call, call.roomId, "busy");
+                this.pause("ringbackAudio");
+                this.play("busyAudio");
+                Modal.createTrackedDialog('Call Handler', 'Call Timeout', ErrorDialog, {
+                    title: _t('Call Timeout'),
+                    description: _t('The remote side failed to pick up') + '.',
+                });
+            } else if (oldState === "invite_sent") {
+                this.setCallState(call, call.roomId, "stop_ringback");
+                this.pause("ringbackAudio");
+            } else if (oldState === "ringing") {
+                this.setCallState(call, call.roomId, "stop_ringing");
+                this.pause("ringbackAudio");
+            } else if (newState === "connected") {
+                this.setCallState(call, call.roomId, "connected");
+                this.pause("ringbackAudio");
+            }
+        });
+    }
+
+    private setCallState(call, roomId, status) {
+        console.log(
+            `Call state in ${roomId} changed to ${status} (${call ? call.call_state : "-"})`,
+        );
+        this.calls[roomId] = call;
+
+        if (status === "ringing") {
+            this.play("ringAudio");
+        } else if (call && call.call_state === "ringing") {
+            this.pause("ringAudio");
+        }
+
+        if (call) {
+            call.call_state = status;
+        }
+        dis.dispatch({
+            action: 'call_state',
+            room_id: roomId,
+            state: status,
+        });
+    }
+
+    private showICEFallbackPrompt() {
+        const cli = MatrixClientPeg.get();
+        const code = sub => <code>{sub}</code>;
+        Modal.createTrackedDialog('No TURN servers', '', QuestionDialog, {
+            title: _t("Call failed due to misconfigured server"),
+            description: <div>
+                <p>{_t(
+                    "Please ask the administrator of your homeserver " +
+                    "(<code>%(homeserverDomain)s</code>) to configure a TURN server in " +
+                    "order for calls to work reliably.",
+                    { homeserverDomain: cli.getDomain() }, { code },
+                )}</p>
+                <p>{_t(
+                    "Alternatively, you can try to use the public server at " +
+                    "<code>turn.matrix.org</code>, but this will not be as reliable, and " +
+                    "it will share your IP address with that server. You can also manage " +
+                    "this in Settings.",
+                    null, { code },
+                )}</p>
+            </div>,
+            button: _t('Try using turn.matrix.org'),
+            cancelButton: _t('OK'),
+            onFinished: (allow) => {
+                SettingsStore.setValue("fallbackICEServerAllowed", null, SettingLevel.DEVICE, allow);
+                cli.setFallbackICEServerAllowed(allow);
+            },
+        }, null, true);
+    }
+
+    private onAction = (payload) => {
```

Comment:
```suggestion
    private onAction = (payload: ActionPayload) => {
```
Will need the import

Reply:
done with import